File: update_tracker.js
import fs from 'fs';
import * as xlsx from 'xlsx';

const trackerPath = 'I:/My Drive/Automation/Shreerang 2026/Change Tracker/Change Tracker.xlsx';

try {
    let workbook;
    if (fs.existsSync(trackerPath)) {
        workbook = xlsx.readFile(trackerPath);
    } else {
        workbook = xlsx.utils.book_new();
        workbook.SheetNames.push('Sheet1');
        workbook.Sheets['Sheet1'] = xlsx.utils.aoa_to_sheet([['Date', 'Feature', 'Description', 'Author', 'Status']]);
    }

    const sheetName = workbook.SheetNames[0];
    const worksheet = workbook.Sheets[sheetName];

    const data = xlsx.utils.sheet_to_json(worksheet, { header: 1 });
    data.push([
        new Date().toLocaleDateString(),
        "Admin Dashboard Redesign",
        "Applied shreerang_dashboard_v10 layout to React components, integrated Tally Prime URL",
        "AI Assistant",
        "Completed"
    ]);

    const newWorksheet = xlsx.utils.aoa_to_sheet(data);
    workbook.Sheets[sheetName] = newWorksheet;
    xlsx.writeFile(workbook, trackerPath);
    console.log('Tracker updated successfully.');
} catch (error) {
    console.error('Error updating tracker:', error.message);
}

File: update_tracker.py
import openpyxl
import datetime
import os
import sys

tracker_path = r'I:\My Drive\Automation\Shreerang 2026\Change Tracker\Change Tracker.xlsx'

try:
    if os.path.exists(tracker_path):
        wb = openpyxl.load_workbook(tracker_path)
        ws = wb.active
    else:
        wb = openpyxl.Workbook()
        ws = wb.active
        ws.append(['Date', 'Feature', 'Description', 'Author', 'Status'])

    today = datetime.datetime.now().strftime("%d-%m-%Y")
    ws.append([
        today, 
        "Admin Dashboard Redesign & Tally Link", 
        "Replicated shreerang_dashboard_v10 layout to React AdminDashboard.jsx & AdminLayout.jsx. Integrated ngrok-free Tally URL.", 
        "AI Assistant", 
        "Completed"
    ])

    wb.save(tracker_path)
    print("Tracker updated successfully using Python.")
except Exception as e:
    print(str(e), file=sys.stderr)
    sys.exit(1)
